sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/TimeExpressionsSuite.scala

```diff
@@ -51,4 +51,37 @@ class TimeExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       condition = "CANNOT_PARSE_TIME",
       parameters = Map("input" -> "'100:50'", "format" -> "'mm:HH'"))
   }
+
+  test("Minute with TIME type") {
+    // A few test times in microseconds since midnight:
+    //   (time in microseconds, expected minute)
+    val testTimes = Seq(
+      (localTime(), 0), // 00:00:00 => 0
+      (localTime(12, 58, 59), 58), // 12:58:59 => 58
+      (localTime(0, 59), 59),   // 00:59:00 => 59
+      (localTime(23, 0, 1), 0), // 23:00:01 => 0
+      (localTime(1), 0), // 01:00:00 => 0
+      (localTime(14, 30), 30), // 14:30:00 => 30
+      (localTime(23, 59, 59, 999999), 59) // 23:59:59.999999 => 59
```

Comment:
The comments look useless. Just do that in the code:
```suggestion
      localTime(23, 59, 59, 999999) -> 59
```